### PR TITLE
Update fetch asset guide to new LSP725.js

### PR DESCRIPTION
**Pull request by fhildeb**
- [x] Setup
- [x] Step 1 - Get all assets ever received
- [x] Step 2 - Get all assets ever issued
- [x] Step 3 - Check ownership of assets
- [x] Step 4 - Check the type of an asset
- [x] Step 5 - Receive the encoded asset data
- [x] Step 6 - Decode the asset data
- [x] Step 7 - Create the storage link
- [x] Step 8 - Get the asset data
- [x] Final Code

 Heading Changes:

- [x] Step 2 - Get all assets ever issued
- [x] Remove Step 9 (fetch properties) as in fetch profile data guide

Still missing:

- [x] Asset URL can not be fetched on new standard
- [x] Add legacy code for Step 2
- [x] Rework comments

### CODE REVIEW

File: docs/guides/fetch-data/read-asset-data.md [hunk 1]
```diff
@@ -37,23 +37,111 @@ Open a terminal in the project's folder of your choice and install all required
...
+  }
+}
+
+// Debug
+fetchUniversalReceiver(SAMPLE_PROFILE_ADDRESS).then((receiverAddress) =>
+  console.log(receiverAddress)
+);
+```
+
+After we got the Universal Receiver address, we can now receive an array of assets from it by calling the `getAllRawValues()` function.
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:139]: _Suggested changes:_

```suggestion
Using the Universal Receiver address, we can now call the `getAllRawValues()` function on this contract to retrieve an array of received assets.
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 2]
```diff
@@ -74,53 +162,146 @@ To make the guide more understandable, we use sample addresses for the Universal
...
 ```
+  </TabItem>
+</Tabs>
+
+## Step 2 - Get all assets ever issued
+
+<Tabs>
+  <TabItem value="Current Standard" label="Current Standard">
+
+The same way, we fetched received assets, we can fetch all ever received assets with the `fetchData('LSP12IssuedAssets[]')` function from the [erc725.js](../../tools/erc725js/getting-started/) library.
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:220]:
```suggestion
The same way, we fetched received assets, we can fetch all ever issued assets with the `fetchData('LSP12IssuedAssets[]')` function from the [erc725.js](../../tools/erc725js/getting-started/) library.
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 3]
```diff
@@ -74,53 +162,146 @@ To make the guide more understandable, we use sample addresses for the Universal
...
+ * Fetch the ever issued assets from 
+ * the Universal Profile
+ *
+ * @param address of the Universal Profile
+ * @return address[] of the issued assets or custom error
+ */
+async function fetchIssuedAssets(address) {
+  try {
+    const profile = new ERC725(erc725schema, address, provider, config);
+    return await (
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:236]: same as above

File: docs/guides/fetch-data/read-asset-data.md [hunk 4]
```diff
@@ -165,118 +392,88 @@ async function getOwnedAddresses(owner) {
...
-  <TabItem value="Current Standads" label="Current Standards">
+  <TabItem value="Current Standards" label="Current Standards">
 
 :::info
 
-Thanks to [function overloading], the `ERC725Y` interface function `getData(...)` can accept:
+By using [function overloading], the `ERC725Y` interface function `getData(...)` can accept:
 
-- either one key: `getData(key)`.
+- either one key: `getData(key)` to fetch a single value.
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:419]:
```suggestion
- either one data key: `getData(key)` to fetch a single value.
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 5]
```diff
@@ -165,118 +392,88 @@ async function getOwnedAddresses(owner) {
...
 - or an array of keys: `getData(keys[])` to fetch multiple values at once.
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:420]:
```suggestion
- or an array of data keys: `getData(keys[])` to fetch multiple values at once.
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 6]
```diff
@@ -165,118 +392,88 @@ async function getOwnedAddresses(owner) {
...
+  <TabItem value="Legacy Standards" label="Legacy Standards">
 
-- in the **legacy** `ERC725Y` interface, `getData(...)` only takes a single key: `getData(key)`.
-- in the **current** `ERC725Y` interface, `getData(...)` can accept:
-  - either one key: `getData(key)`
-  - or an array of keys: `getData(keys[])`
+:::info
 
-The current standard therefore allows to fetch multiple values at once, via [function overloading].
+While using the **legacy** `ERC725Y` interface, `getData(...)` only takes one single key: `getData(key)`.
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:469]:
```suggestion
While using the **legacy** `ERC725Y` interface, `getData(...)` only takes one single data key: `getData(key)`.
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 7]
```diff
@@ -303,168 +500,108 @@ The current standard therefore allows to fetch multiple values at once, via [fun
...
 );
 ```
 
   </TabItem>
 </Tabs>
 
-## Step 4 - Receive the encoded asset data
+## Step 5 - Receive the encoded asset data
+
+Now we can safely call the data of the address. The [LSP4](../../standards/nft-2.0/LSP4-Digital-Asset-Metadata) data is saved in a ERC725Y key-value store, and we need to input the right key to fetch the associated value. There are multiple [LSP4 keys](../../standards/nft-2.0/LSP4-Digital-Asset-Metadata) for different properties.
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:546]:
```suggestion
Now we can safely retrieve the metadata for the asset address. The [LSP4](../../standards/nft-2.0/LSP4-Digital-Asset-Metadata) metadata is saved under the ERC725Y key-value store of the digital asset. We need to input the correct data key to fetch the associated value. There are multiple [LSP4 keys](../../standards/nft-2.0/LSP4-Digital-Asset-Metadata) for different properties.
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 8]
```diff
@@ -303,168 +500,108 @@ The current standard therefore allows to fetch multiple values at once, via [fun
...
+
+Now we can safely call the data of the address. The [LSP4](../../standards/nft-2.0/LSP4-Digital-Asset-Metadata) data is saved in a ERC725Y key-value store, and we need to input the right key to fetch the associated value. There are multiple [LSP4 keys](../../standards/nft-2.0/LSP4-Digital-Asset-Metadata) for different properties. 
 
+- `LSP4TokenName`
+- `LSP4TokenSymbol`
+- `LSP4Metadata`
+- `LSP4Creators[]`
 
-Now we can safely call the data of the address. The [LSP4](../../standards/nft-2.0/LSP4-Digital-Asset-Metadata) data is saved in a ERC725Y key-value store, and we need to input the right key to fetch the associated value. There are multiple keys for different properties. To give a showcase, we will use the metadata key to receive the associated data.
+To give a showcase, we will use the `LSP4Metadata` key to receive the associated data.
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:553]:
```suggestion
In this guide, we will use the `LSP4Metadata` key to read the asset metadata.
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 9]
```diff
@@ -303,168 +500,108 @@ The current standard therefore allows to fetch multiple values at once, via [fun
...
 // ...
-// ABI's
-const ERC725ABI = require('@erc725/smart-contracts/artifacts/ERC725.json');
-const LSP4schema = require('./lsp4_schema.json');
 
-// Keys for properties
+// ABIs
+const LSP4schema = require('@erc725/erc725.js/schemas/LSP4DigitalAsset.json');
+
+// Keys for asset properties
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:566]:
```suggestion
// Data keys for asset properties
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 10]
```diff
@@ -303,168 +500,108 @@ The current standard therefore allows to fetch multiple values at once, via [fun
...
 const CreatorsKey = LSP4schema[4].key;
 
 /*
- * Fetch the dataset of an asset
- * from the Universal Profile
+ * Get the dataset of an asset
  *
- * @param key of asset property
- * @return string of encoded data
+ * @param key of the property to fetch
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:575]:
```suggestion
 * @param data key of the property to fetch
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 11]
```diff
@@ -490,222 +627,155 @@ getAssetData(MetaDataKey).then((encodedData) => console.log(encodedData));
...
-const ERC725LegacyInterface = require('./erc725_legacy_interface.json');
-const ERC725ABI = require('@erc725/smart-contracts/artifacts/ERC725.json');
-const LSP4schema = require('./lsp4_schema.json');
 
-// Keys for properties
+// ABIs
+const ERC725MinimalABI = require("./erc725_legacy_minimal_abi.json");
+const LSP4schema = require("@erc725/erc725.js/schemas/LSP4DigitalAsset.json");
+
+// Keys for asset properties
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:638]:
```suggestion
// Data keys for asset properties
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 12]
```diff
@@ -490,222 +627,155 @@ getAssetData(MetaDataKey).then((encodedData) => console.log(encodedData));
...
 const CreatorsKey = LSP4schema[4].key;
 
 /*
- * Fetch the dataset of an asset
- * from the Universal Profile
+ * Get the dataset of an asset
  *
- * @param key of asset property
- * @return string of encoded data
+ * @param key of the property to fetch
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:647]:
```suggestion
 * @param Data key of the property to fetch
```

File: docs/guides/fetch-data/read-asset-data.md [hunk 13]
```diff
@@ -490,222 +627,155 @@ getAssetData(MetaDataKey).then((encodedData) => console.log(encodedData));
...
-const decodeTokenPhrase = LSP4schema[2].name;
-const decodeMetaPhrase = LSP4schema[3].name;
-const decodeCreatorPhrase = LSP4schema[4].name;
+// ...
 
 /*
- * Decode value from ERC725Y storage
- * based on it's key and phrase
+ * Decode the value from ERC725Y storage
+ * based on its key and phrase
```
Comment (CJ42) [inline docs/guides/fetch-data/read-asset-data.md:689]:
```suggestion
 * based on its data key and phrase
```

**Review by CJ42 — COMMENTED** [review]
Let's make sure that we replace the term `key` by `data key` in the guide.

**Comment by fhildeb** [comment]
> Let's make sure that we replace the term `key` by `data key` in the guide.

In general within the text, or text & code snippets?

File: docs/guides/fetch-data/read-asset-data.md [hunk 14]
```diff
@@ -490,222 +627,155 @@ getAssetData(MetaDataKey).then((encodedData) => console.log(encodedData));
...
 
 </Tabs>
 
-## Step 5 - Decode the asset data
+## Step 6 - Decode the asset data
 
 We can now decode the encoded metadata to fetch readable information. We use
-`decodeData()` from the `erc725.js` library. While using ERC725, we will have
-to declare a config and provider as we did while [reading profile data](./read-profile-data).
+the `decodeData()` function from the [`erc725.js`](../../tools/erc725js/getting-started/) library. We will continue the step before and showcase decoding the `LSP4Metadata` key.
```
[the remark this answers is not in the document]
Comment (fhildeb) [reply docs/guides/fetch-data/read-asset-data.md:680]: changed to this link (but changed `//` --> `/`)

**Review by Hugoo — APPROVED** [review]

**Pull request merged**